### ISSUE TITLE
"Your First App" tutorial should show complete code context

### ISSUE THREAD

**Issue by liamdebeasi** [ionic-team/ionic-docs#2082]
**URL**
https://ionicframework.com/docs/angular/your-first-app#photo-gallery-

**What is missing or inaccurate about the content on this page?**

The tutorial should show the complete code blocks so that it is easy to copy and paste. Right now, the code blocks only show what you need to add or change, but it does not show where the code should go or any additional context. For more experienced Ionic devs this may be fine, but for those starting out with Ionic I think this additional context would go a long way in making this tutorial more approachable.

We received feedback on this here: https://forum.ionicframework.com/t/angular-build-your-first-app-tutorial-suggestion/216065

**Comment by damascene** [comment]
I agree, it's very important otherwise user may think that the documentation are outdated. I'm just testing ionic for the first time and I find it difficult to put code blocks in the correct order to have taste of how this works.
